### ISSUE TITLE
Memory leak with static canvas in nodejs

### ISSUE THREAD

**Issue by LaleWolf** [fabricjs/fabric.js#6125]
## Version
fabric: **3.6.1**
jsdom: **15.2.1**
node: **8.16.0**

## Test Case
https://gist.github.com/LaleWolf/a654a40fbb89247abefe7a71f253aace#file-fabricmemoryleak-js

## Information about environment
node: **8.16.0**

## Steps to reproduce
[Download following gist as ](https://gist.github.com/LaleWolf/a654a40fbb89247abefe7a71f253aace#file-fabricmemoryleak-js) and start with `node --expose-gc .\fabricMemoryLeak.js`

### Description
Creating a `StaticCanvas` leads to a memory leak. Allthough canvas is created with `objectCaching = false` and `enableRetinaScaling: false` options and its getting cleared and disposed after creation.

To visualize the issue, in an inner loop (**1k times**) a `StaticCanvas` is getting created, filled with a simple `fabric.Rect` and cleared and disposed afterwards. 
An outer loop (**10 times**) is logging the memoryUsage of node and calling the garbage collector.

## Expected Behavior
[Following gist](https://gist.github.com/LaleWolf/51cf9be735c33a946bc2a9ede0d9f37e#file-jsdomonlyfabricmemoryleak-js) (start with `node --expose-gc .\jsdomOnlyFabricMemoryLeak.js`) represents the same use case with JSDOM only. The total heap size is staying stable, the used heap is going up and down again as expected.

```
{ rss: 70877184,        heapTotal: 59076608,    heapUsed: 33445296,     external: 1593690 }
{ rss: 74661888,        heapTotal: 62222336,    heapUsed: 23266016,     external: 180082958 }
{ rss: 74850304,        heapTotal: 62222336,    heapUsed: 23244120,     external: 360082958 }
{ rss: 74940416,        heapTotal: 62222336,    heapUsed: 23253488,     external: 540082958 }
{ rss: 75112448,        heapTotal: 62222336,    heapUsed: 23275936,     external: 720082958 }
{ rss: 75071488,        heapTotal: 62222336,    heapUsed: 23270240,     external: 900082958 }
{ rss: 75198464,        heapTotal: 62222336,    heapUsed: 23323728,     external: 1080082958 }
{ rss: 75309056,        heapTotal: 62222336,    heapUsed: 23320128,     external: 1260082958 }
{ rss: 75280384,        heapTotal: 62222336,    heapUsed: 23314488,     external: 1440082958 }
{ rss: 75538432,        heapTotal: 62222336,    heapUsed: 23322392,     external: 1620082958 }
{ rss: 75603968,        heapTotal: 62222336,    heapUsed: 23313184,     external: 1800082958 }
```

## Actual Behavior
The total and used heap is growing over time.

```
{ rss: 81424384,        heapTotal: 63393792,    heapUsed: 42636960,     external: 2394489 }
{ rss: 95371264,        heapTotal: 67788800,    heapUsed: 28668616,     external: 180622958 }
{ rss: 96882688,        heapTotal: 68837376,    heapUsed: 29068272,     external: 360622958 }
{ rss: 98283520,        heapTotal: 69885952,    heapUsed: 29418992,     external: 540622958 }
{ rss: 99430400,        heapTotal: 69885952,    heapUsed: 29759688,     external: 720622958 }
{ rss: 100069376,       heapTotal: 70410240,    heapUsed: 30002776,     external: 900622958 }
{ rss: 112521216,       heapTotal: 70410240,    heapUsed: 30401128,     external: 1080622958 }
{ rss: 112971776,       heapTotal: 70410240,    heapUsed: 30620528,     external: 1260622958 }
{ rss: 102060032,       heapTotal: 70410240,    heapUsed: 30984720,     external: 1440622958 }
{ rss: 114507776,       heapTotal: 70934528,    heapUsed: 31233760,     external: 1620622958 }
{ rss: 114892800,       heapTotal: 71983104,    heapUsed: 31621712,     external: 1800622958 }
```

**Comment by asturur** [comment]
how much are we leaking? 1 around 1 MB per 1000 iterations? 1kb per iteration?

Just to understand more or less what to look for.

**Comment by LaleWolf** [comment]
I went through some analisys to give you better information.
Overall after 60x1000 iterations, we can say we are leaking 360KB per 1000 iterations on average just for creating a `StaticCanvas`.

Thats the memory usage over time. Started with around 100MB after first iteration (1000 done) and ended with around 160MB after last iteration (60000 done).
![memory usage (1)](https://user-images.githubusercontent.com/2292243/73469767-2d6d3880-4387-11ea-9624-4219d1df8626.png)

**Comment by LaleWolf** [comment]
Just wanted to know if you have any ideas/updates on this topic. Let me know if you need any other informations or anything I can help.

**Comment by asturur** [comment]
no, my only idea is following line by line the process of creating and dismissing a static canvas and look for something that gets attached to the main fabricObject and not cleaned for example.
Not much more than that.

**Comment by gloriousjob** [comment]
I tried this with fabric 4.2.0 and got:
{ rss: 77164544,        heapTotal: 55173120,    heapUsed: 35609720,     external: 3309927 }
{ rss: 92180480,        heapTotal: 60764160,    heapUsed: 23698112,     external: 1551892 }
{ rss: 92504064,        heapTotal: 61026304,    heapUsed: 23805536,     external: 1551892 }
{ rss: 92364800,        heapTotal: 61288448,    heapUsed: 24053288,     external: 1551892 }
{ rss: 94957568,        heapTotal: 61288448,    heapUsed: 23931760,     external: 1551892 }
{ rss: 93138944,        heapTotal: 61026304,    heapUsed: 23772056,     external: 1551892 }
{ rss: 105041920,       heapTotal: 61288448,    heapUsed: 23924312,     external: 1551892 }
{ rss: 105123840,       heapTotal: 60764160,    heapUsed: 23793936,     external: 1551892 }
{ rss: 93016064,        heapTotal: 60764160,    heapUsed: 23846064,     external: 1551892 }
{ rss: 105660416,       heapTotal: 61026304,    heapUsed: 23989624,     external: 1551892 }
{ rss: 106897408,       heapTotal: 60764160,    heapUsed: 23870712,     external: 1551892 }

Just to see, I also tried fabric 3.6.1:
{ rss: 77172736,        heapTotal: 55128064,    heapUsed: 35119008,     external: 3301521 }
{ rss: 92811264,        heapTotal: 60981248,    heapUsed: 23658560,     external: 1551892 }
{ rss: 93048832,        heapTotal: 60981248,    heapUsed: 23746312,     external: 1551892 }
{ rss: 94183424,        heapTotal: 61243392,    heapUsed: 24011176,     external: 1551892 }
{ rss: 94642176,        heapTotal: 60981248,    heapUsed: 23876248,     external: 1551892 }
{ rss: 93155328,        heapTotal: 60719104,    heapUsed: 23770552,     external: 1551892 }
{ rss: 105185280,       heapTotal: 61243392,    heapUsed: 24120112,     external: 1551892 }
{ rss: 105631744,       heapTotal: 60981248,    heapUsed: 23851712,     external: 1551892 }
{ rss: 95219712,        heapTotal: 60981248,    heapUsed: 23890128,     external: 1551892 }
{ rss: 106496000,       heapTotal: 61243392,    heapUsed: 24028456,     external: 1551892 }
{ rss: 105283584,       heapTotal: 60981248,    heapUsed: 23871480,     external: 1551892 }

I am using node 12.18.3 so not sure if that is making a difference. The used and totals seem to be hovering around the same numbers for me.
